### DISCUSSION TITLE
[Question]: data-provider modify

### DISCUSSION THREAD

**Discussion by kuangxiaoye** [danny-avila/LibreChat#2466] [Q&A]
### What is your question?

After modifying file schemas.d.ts, executing command npm run frontend will force the changes made to file schemas.d.ts to be reverted.

### More Details

I want to modify TMessage field and add some content during secondary development, but I cannot modify it.

### Screenshots

![image](https://github.com/danny-avila/LibreChat/assets/43713316/ece53bdb-9e5c-4d44-94fa-b446962cdf3a)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
> schemas.d.ts

This is not the file to modify. files with `.d.` in extension are transient and will be discarded upon re-compiling data-provider code.

The file you should edit is `packages/data-provider/src/schemas.ts`


See the contributing docs: https://docs.librechat.ai/contributions/index.html

**Reply by kuangxiaoye** [reply]
Thank you very much for your answer! This solved my problem!
